subsys/usb/device_next/class/usbd_midi.c

```diff
@@ -0,0 +1,752 @@
+	BUILD_ASSERT(DT_REG_ADDR(node) + DT_REG_SIZE(node) <= 16,          \
+		     "Too many Group Terminals in this Block");
+
+#define USBD_MIDI_VALIDATE_INSTANCE(inst) \
+	DT_INST_FOREACH_CHILD(inst, USBD_MIDI_VALIDATE_GRPTRM_BLOCK)
+
+#define USBD_MIDI2_INIT_GRPTRM_BLOCK_DESCRIPTOR(node)                       \
+	{                                                                   \
+		.bLength = sizeof(struct usb_midi_grptrm_block_descriptor), \
+		.bDescriptorType = CS_GR_TRM_BLOCK,                         \
+		.bDescriptorSubtype = GR_TRM_BLOCK,                         \
+		.bGrpTrmBlkID = GRPTRM_BLOCK_ID(node),                      \
+		.bGrpTrmBlkType = GRPTRM_BLOCK_TYPE(node),                  \
+		.nGroupTrm = DT_REG_ADDR(node),                             \
+		.nNumGroupTrm = DT_REG_SIZE(node),                          \
+		.iBlockItem = 0,                                            \
+		.bMIDIProtocol = GRPTRM_PROTOCOL(node),                     \
+		.wMaxInputBandwidth = 0x0000,                               \
+		.wMaxOutputBandwidth = 0x0000,                              \
+	}
+
+#define USBD_MIDI2_GRPTRM_TOTAL_LEN(inst)                 \
+	sizeof(struct usb_midi_grptrm_header_descriptor)  \
+	+ DT_INST_CHILD_NUM_STATUS_OKAY(inst)             \
+	* sizeof(struct usb_midi_grptrm_block_descriptor)
+
+#define USBD_MIDI_DEFINE_DESCRIPTORS(inst)                                               \
+	static struct usbd_midi_descriptors usbd_midi_desc_##inst = {                    \
+		.iad = {                                                                 \
+			.bLength = sizeof(struct usb_association_descriptor),            \
+			.bDescriptorType = USB_DESC_INTERFACE_ASSOC,                     \
+			.bFirstInterface = 0,                                            \
+			.bInterfaceCount = 2,                                            \
+			.bFunctionClass = AUDIO,                                         \
+			.bFunctionSubClass = MIDISTREAMING,                              \
+		},                                                                       \
+		.if0_std = {                                                             \
+			.bLength = sizeof(struct usb_if_descriptor),                     \
+			.bDescriptorType = USB_DESC_INTERFACE,                           \
+			.bInterfaceNumber = 0,                                           \
+			.bAlternateSetting = 0,                                          \
+			.bNumEndpoints = 0,                                              \
+			.bInterfaceClass = AUDIO,                                        \
+			.bInterfaceSubClass = AUDIOCONTROL,                              \
+		},                                                                       \
+		.if0_cs = {                                                              \
+			.bLength = sizeof(struct usb_midi_cs_ac_header_descriptor),      \
+			.bDescriptorType = CS_INTERFACE,                                 \
```

Comment:
There are USB_DESC_CS_INTERFACE/USB_DESC_CS_ENDPOINT, they should be used instead.